src/System.Xml.XmlSerializer/tests/Performance/XsPerformanceTest.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.IO;
+using System.Runtime.Serialization;
+
+namespace System.Xml.XmlSerializer.Tests.Performance
+{
+    internal class XsSerializerFactory : SerializerFactory
+    {
+        private static XsSerializerFactory _instance = null;
+        public static XsSerializerFactory GetInstance()
+        {
+            if (_instance == null)
+            {
+                _instance = new XsSerializerFactory();
+            }
+            return _instance;
+        }
+
+        public override IPerfTestSerializer GetSerializer(SerializerType serializerType)
+        {
+            switch (serializerType)
+            {
+                case SerializerType.XmlSerializer:
+                    return new XsSerializer();
+                default:
+                    throw new NotImplementedException();
```

Comment:
This doesn't seem to be the right exception to throw from here. Maybe NotSupportedException?